### PR TITLE
fix(files): Use DAV functions from library to fix invalid source links

### LINKED ISSUE #42333: [Bug]: NC28 Recent files have incorrect URL
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

In Files and navigating to the "Recent" page. The links are broken where as an example file (test.txt) is severed up with the following URL:

`https://nextcloud-instance.com/remote.php/dav/files/admin/files/admin/test.txt`

This URL is wrong the file does not exist there and you are navigated to a page where no files exist. The URL should be:

`https://nextcloud-instance.com/remote.php/dav/files/admin/test.txt`

Note the duplicate of `/files/admin` in the URL string.
I'm not sure related but the banner of recent files at the top of the All Files page also results in the links not working but the behavior is different...You click a file listed there and nothing happens.

### Steps to reproduce

1. Open the Recent section of the Files page
2. Click on a recent file listed
3. You are navigated to a page which the file does not exist and therefore nothing opens.

### Expected behavior

Clicking a file from the recent page should open the file.

### Installation method

Community Docker image

### Nextcloud Server version

28

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 22 to 23)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "objectstore": {
            "class": "\\OC\\Files\\ObjectStore\\S3",
            "arguments": {
                "bucket": "***REMOVED SENSITIVE VALUE***",
                "hostname": "s3.us-west-1.wasabisys.com",
                "key": "***REMOVED SENSITIVE VALUE***",
                "secret": "***REMOVED SENSITIVE VALUE***",
                "port": 443,
                "region": "us-west-1",
                "use_ssl": true
            }
        },
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "28.0.0.11",
        "overwrite.cli.url": "https:\/\/***REMOVED SENSITIVE VALUE***",
        "overwriteprotocol": "https",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_smtpsecure": "tls",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "filelocking.enabled": true,
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "password": "***REMOVED SENSITIVE VALUE***",
            "timeout": 0,
            "port": 6379
        },
        "default_phone_region": "US",
        "simpleSignUpLink.shown": false,
        "skeletondirectory": "",
        "logtimezone": "America\/Phoenix",
        "app_install_overwrite": [
            "breezedark",
            "backup",
            "sharerenamer",
            "files_photospheres",
            "metadata",
            "drawio",
            "files_rightclick",
            "cfg_share_links"
        ],
        "enforce_theme": "",
        "maintenance": false,
        "preview_max_x": "2048",
        "preview_max_y": "2048",
        "jpeg_quality": "60",
        "loglevel": 2
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.20.0
  - cfg_share_links: 4.2.0
  - cloud_federation_api: 1.11.0
  - comments: 1.18.0
  - dashboard: 7.8.0
  - dav: 1.29.1
  - deck: 1.12.0
  - federatedfilesharing: 1.18.0
  - federation: 1.18.0
  - files: 2.0.0
  - files_photospheres: 1.27.0
  - files_reminders: 1.1.0
  - files_sharing: 1.20.0
  - files_trashbin: 1.18.0
  - files_versions: 1.21.0
  - groupfolders: 16.0.1
  - logreader: 2.13.0
  - lookup_server_connector: 1.16.0
  - metadata: 0.19.0
  - notifications: 2.16.0
  - oauth2: 1.16.3
  - onlyoffice: 9.0.0
  - password_policy: 1.18.0
  - photos: 2.4.0
  - previewgenerator: 5.4.0
  - privacy: 1.12.0
  - provisioning_api: 1.18.0
  - recommendations: 2.0.0
  - related_resources: 1.3.0
  - serverinfo: 1.18.0
  - settings: 1.10.0
  - sharebymail: 1.18.0
  - spreed: 18.0.1
  - systemtags: 1.18.0
  - text: 3.9.1
  - theming: 2.3.0
  - theming_customcss: 1.15.0
  - twofactor_backupcodes: 1.17.0
  - updatenotification: 1.18.0
  - viewer: 2.2.0
  - workflowengine: 2.10.0
Disabled:
  - admin_audit: 1.18.0
  - bruteforcesettings: 2.8.0
  - circles: 28.0.0-dev (installed 27.0.0)
  - contactsinteraction: 1.9.0 (installed 1.8.0)
  - drawio: 2.1.4 (installed 2.1.4)
  - encryption: 2.16.0
  - files_external: 1.20.0 (installed 1.19.0)
  - files_pdfviewer: 2.9.0 (installed 2.8.0)
  - firstrunwizard: 2.17.0 (installed 2.16.0)
  - nextcloud_announcements: 1.17.0 (installed 1.16.0)
  - support: 1.11.0 (installed 1.10.0)
  - survey_client: 1.16.0 (installed 1.15.0)
  - suspicious_login: 6.0.0
  - twofactor_totp: 10.0.0-beta.2
  - user_ldap: 1.19.0
  - user_saml: 6.0.1 (installed 6.0.1)
  - user_status: 1.8.1 (installed 1.7.0)
  - weather_status: 1.8.0 (installed 1.7.0)
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
Can be provided if absolutely necessary.
```

### Additional info

Tried in both Chrome and Edge

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: https://github.com/nextcloud/server/issues/42333

## Summary
Refactor the code to use the files library functions instead which will fix invalid source values causing #4233 

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by artonge — APPROVED** [review]

**Comment by artonge** [comment]
Let's wait for @skjnldsv approval

**Review by emoral435 — APPROVED** [review]
Tested on my local dev env - works now ! Still agree with Louis that we should wait for @skjnldsv approval

**Review by skjnldsv — APPROVED** [review]

File: apps/files/src/services/Recent.ts [hunk 1]
```diff
@@ -112,11 +48,11 @@ export const getContents = async (path = '/'): Promise<ContentsWithRoot> => {
...
 			id: 0,
-			source: generateRemoteUrl('dav' + rootPath),
-			root: rootPath,
+			source: `${davRemoteURL}${davRootPath}`,
+			root: davRootPath,
 			owner: getCurrentUser()?.uid || null,
 			permissions: Permission.READ,
 		}),
-		contents: contents.map(resultToNode),
+		contents: contents.map((r) => davResultToNode(r)),
```
Comment (skjnldsv) [inline apps/files/src/services/Recent.ts:56]: Why not ```contents: contents.map(davResultToNode)``` ?
(not a blocker, simply curious))

Comment (susnux) [reply apps/files/src/services/Recent.ts:56]: Parameters do not match `map` has a signature of `element: T, index: number` but `davResultToNode` has `element: Result, root?: string` as it allows for different root path.

**Pull request merged**